c10/core/TensorImpl.h

```diff
@@ -224,6 +225,27 @@ struct C10_API NamedTensorMetaInterface {
   };
 };
 
+struct C10_API ExtraMeta {
+  SymDimVector sizes_ = {0};
+  SymDimVector strides_ = {1};
+  SymInt numel_ = 1;
+  SymInt storage_offset_ = 0; // TODO
+  // TODO:
+  // SymBool is_contiguous_;
+  std::unique_ptr<c10::NamedTensorMetaInterface> named_tensor_meta_ = nullptr;
+
+  std::unique_ptr<ExtraMeta> clone() const {
```

Comment:
nit: adding a constructor would likely be more efficient here. haven't checked assembly but I bet we either actually perform at least one default construction or fail to optimize out at least one is_symbolic check; even if not, it's cheap to be nice to the poor overworked inliner here